### PR TITLE
Rewrite staff_hours functions for new schema from ocf/etc.

### PR DESCRIPTION
**Pull request by redplanks**
Depends on ocf/etc#28

### CODE REVIEW

File: ocflib/lab/staff_hours.py [hunk 1]
```diff
@@ -30,40 +28,61 @@ def gravatar(self, size=100):
...
-            ],
-            cancelled=hour['cancelled'],
-        ) for hour in staff_hours['staff-hours']
-    ]
+    staff_hour_list = []
+    # If we try iterating through the keys in staff-hours, it's non-deterministic ordering.
+    # Because of the schema, we are guaranteed that these days are always here.
+    for day in ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']:
+        if not staff_hours['staff-hours'][day]:
+            continue
```
Comment (dkess) [inline ocflib/lab/staff_hours.py]: This line isn't necessary, since each day is guaranteed to be present.

File: ocflib/lab/staff_hours.py [hunk 2]
```diff
@@ -30,40 +28,61 @@ def gravatar(self, size=100):
...
+
+        for staff_hour in staff_hours['staff-hours'][day]:
+            cancelled = False
+            if 'cancelled' in staff_hour.keys():
+                cancelled = staff_hour['cancelled']
```
Comment (dkess) [inline ocflib/lab/staff_hours.py]: This can be shortened to `staff_hour.get('cancelled', False)`

File: ocflib/lab/staff_hours.py (hunk 1 above)
Comment (redplanks) [reply ocflib/lab/staff_hours.py]: It's an ugly workaround, but if I don't check days in increasing lexical order, then the order of `Hour`s returned is non-determinstic, which breaks the tests. I guess I could use a set in the test?

**Comment by dkess** [comment]
I'm saying you don't get the conditional that checks if each day of the week is present, since it's guaranteed to be present. You can still use this kind of loop.

**Comment by redplanks** [comment]
I need the check since if the day doesn't have any entries under it, the dict entry will be `None`.

**Pull request merged**